tdm/docs/architecture.md

```diff
@@ -0,0 +1,262 @@
+   support the former, while it relies on the Attestation Agent API to support
+   the latter case. It provides attestation results back to the TDM core.
+
+```
+
+                      ┌────────────────────────┐
+                      │ Trusted Device Manager │
+      ┌───────────────┴────────────────────────┴────────────┐
+      │                                     ┌──────────┐    │
+      │                                     │Reference │    │
+      │       ┌───────────┐ ┌───────────┐ ┌─│  Values  │    │
+      │       │  Device   │ │  Device   │ │ └──────────┘    │
+   ┏━━╋━━━━━━▶│ Detection │ │Attestation│◁┤ ┌───────────┐   │
+   ┃  │       │  Module   │ │  Module   │ │ │Attestation│   │
+   ┃  │       └───────────┘ └───────────┘ └─│ Policies  │   │
+   ┃  │             △             △         └───────────┘   │
+   ┃  │             └──────┐┌─────┘                         │
+   ┃  │                    ▽▽                               │
+   ┃  │             ┌──────────────┐                        │
+   ┃  │             │              │                        │
+   ┃  │             │ Core Module  │                        │
+   ┃  │             │              │                        │
+   ┃  │             └──────────────┘                        │
+   ┃  │                     △                               │
+ PCIe │                     └────┐                          │
+ udev │                          │                          │
+events│                          ▽                          │
+   ┃  │         ┌─────────────────────────────────┐         │
+   ┃  │         │       ABI Plugins Module        │         │
+   ┃  │         └─────────────────────────────────┘         │
+   ┃  │         ┌──────────┐┌──────────┐┌─────────┐         │
+   ┃  │         │TDX Plugin││SEV Plugin││SE Plugin│         │
+   ┃  │         └──────────┘└──────────┘└─────────┘         │
+   ┃  │                          ┃                          │
+   ┃  └──────────────────────────╋──────────────────────────┘
+   ┃                             ┃
+   ┃                         CoCo ABI
+   ┃                             ┃
+   ┃                             ▼
+   ┃  ┌─────────────────────────────────────────────────────┐
+   ┗━━│                 Linux Guest Kernel                  │
+      └─────────────────────────────────────────────────────┘
+```
+
+### TDM Core
+
+The TDM core module interacts with three other TDM components:
+
+1. The device detection module. It monitors udev events to detect PCI devices
+   addition and removal from the guest emulated PCI bus. This module calls into
+   the TDM core to notify it about both event types. The TDM core then relies on
+   the following two modules to verify, attest and accept newly detected
+   devices. Only TDM-accepted TDIs are visible and available to the rest of the
+   guest software stack.
+2. The ABI plugins module. This is an abstraction over the Linux CoCo
```

Comment:
I feel it might bring some confusing if we call "ABI plugins module".
Maybe just name it as its functionality, e.g.
* Device evidence collection module. (cert chain + measurement)
* TDI state management module.
* TDI Report validation module.

Reply:
What about TDI management module?

Reply:
Sure, any concrete name is better than "plug in module".

Reply:
Fixed then.